mlir/include/mlir/Analysis/Presburger/Fraction.h

```diff
@@ -95,8 +99,44 @@ inline bool operator>=(const Fraction &x, const Fraction &y) {
   return compare(x, y) >= 0;
 }
 
+inline Fraction reduce(const Fraction &f) {
+  if (f == Fraction(0))
+    return f;
+  MPInt g = gcd(f.num, f.den);
+  return Fraction(f.num / g, f.den / g);
+}
+
 inline Fraction operator*(const Fraction &x, const Fraction &y) {
-  return Fraction(x.num * y.num, x.den * y.den);
+  return reduce(Fraction(x.num * y.num, x.den * y.den));
+}
+
+inline Fraction operator/(const Fraction &x, const Fraction &y) {
+  return reduce(Fraction(x.num * y.den, x.den * y.num));
+}
+
+inline Fraction operator+(const Fraction &x, const Fraction &y) {
+  return reduce(Fraction(x.num * y.den + x.den * y.num, x.den * y.den));
+}
+
+inline Fraction operator-(const Fraction &x, const Fraction &y) {
+  return reduce(Fraction(x.num * y.den - x.den * y.num, x.den * y.den));
+}
+
+inline Fraction& operator+=(const Fraction &g, const Fraction &f) {
+  Fraction *r = NULL;
+  *r = g+f;
+  return *r;
+}
```

Comment:
`operator+=` should modify the first argument `g`, it can't be const in `g`! Also, please keep the same argument names x, y for uniformity

```
inline Fraction& operator+=(Fraction &x, const Fraction &y) {
  x = x + y;
  return x;
}
```